scripts/ci/test_plan.py

```diff
@@ -4,23 +4,133 @@
 
 # A script to generate twister options based on modified files.
 
-import re, os
+import re
+import os
 import argparse
 import glob
-import yaml
 import json
 import fnmatch
 import subprocess
 import csv
+from pathlib import Path
 import logging
+import concurrent.futures
+import yaml
+
 from git import Repo
 
+
+
 if "ZEPHYR_BASE" not in os.environ:
     exit("$ZEPHYR_BASE environment variable undefined.")
 
 repository_path = os.environ['ZEPHYR_BASE']
 logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
 
+def _procee_maintainer_file(node, zephyr_base_dir, target_files):
+    ''' procee_maintainer_file
+
+        :param node: items to run in multi-process
+        :type node: Hash
+
+        :param zephyr_base_dir: zephyr base dir
+        :type zephyr_base_dir: string
+
+        :param target_files: target files
+        :type target_files: array
+
+        :return Array of matched application path, otherwise empty array
+    '''
+    matched = False
+    test_apps_path = []
+
+    if "files" in node:
+        _test_apps_path = []
+        for item in node["files"]:
+            logging.info(item)
+            if re.match('^tests', item):
+                _test_apps_path.append(item)
+                continue
+            if re.match('^samples', item):
+                _test_apps_path.append(item)
+                continue
```

Comment:
please reduce repetitiveness

Reply:
> How did you test it? I would like to see some demo. I have troubles understanding what your code is doing. It starts with descriptive names, but down the line it ends with `if [ e for e in __f if e in target_files ]:`. It would be also easier to review if you'd add some comments explaining what the code is trying to achieve. E.g. I think I spent quite some time (too much) trying to figure out what L47-72 is doing

ok, I will add some comments to exaplin the purpose, and I write a standalone code to test this function before I put the code into ci.